src/inference/elbo.ad.js

```diff
@@ -208,6 +209,54 @@ module.exports = function(env) {
       return k(s);
     },
 
+    mapDataFetch: function(data, batchSize, address) {
+
+      var ix;
+      if (batchSize === data.length) {
+        // Use all the data, in order.
+        ix = [];
+      } else {
+        ix = _.times(batchSize, function() {
+          return Math.floor(util.random() * data.length);
```

Comment:
Thanks for the reference, I'd not come across that result before. I went with "with replacement" for simplicity, and for the data set I've been using it with (mnist) both strategies amount to about the same thing. Having "without replacement" may well be worth while (and is on my list of things to look at), but in the interests of getting to some of the other things we want for 0.9 can we merge this pr without it?